### ISSUE TITLE
Nix plugin loads but has no effect on macOS

### ISSUE THREAD

**Issue by 3noch** [NixOS/nix#2189]
```
$ nixVar=$(nix-build -E '(import ((import <nixpkgs> {}).fetchzip { url = "https://d3g5gsiof5omrk.cloudfront.net/nixos/18.03/nixos-18.03.132507.5f2da7f837e/nixexprs.tar.xz"; sha256 = "1f9wyhwdrwzkx3fx63ds3dfqw1n5hl4y6gi9gfaxk2gml0wdy4yp"; }) {}).nix' --no-out-link)

$ nix-shell -p $nixVar
[nix-shell] $ nixPlugins=$(nix-build -E '(import ((import <nixpkgs> {}).fetchzip { url = "https://d3g5gsiof5omrk.cloudfront.net/nixos/18.03/nixos-18.03.132507.5f2da7f837e/nixexprs.tar.xz"; sha256 = "1f9wyhwdrwzkx3fx63ds3dfqw1n5hl4y6gi9gfaxk2gml0wdy4yp"; }) {}).nix-plugins_4' --no-out-link)
[nix-shell] $ nix show-config --plugin-files $nixPlugins/lib/nix/plugins | grep plugin-files
# ... properly registered

[nix-shell] $ nix eval '(builtins.extraBuiltins == null)' --plugin-files $nixPlugins/lib/nix/plugins
```

On NixOS the last command produces `true`. On macOS it produces `error: attribute 'extraBuiltins' missing, at (string):1:2`

**Comment by 3noch** [comment]
cc @shlevy @grahamc

**Comment by shlevy** [comment]
@3noch Thanks, will take a look. Just to check, what's your macOS version?

**Comment by 3noch** [comment]
macOS High Sierra 10.13.4

**Comment by shlevy** [comment]
@3noch nix-plugins_4 is linked against nixUnstable. If I change your first command to use `nixUnstable` it all works as expected.

**Comment by shlevy** [comment]
I'll check to see if it builds against nix stable now.

**Comment by 3noch** [comment]
@shlevy Why is it linked against a different nixpkgs than where it comes from?

**Comment by 3noch** [comment]
And why would there be a difference between macOS and NixOS?

**Comment by shlevy** [comment]
It's not different nixpkgs, it's `nixUnstable` vs `nix`. I'm not sure why it would be different on Linux, except pleasant coincidence.

**Comment by 3noch** [comment]
@shlevy Ah right. It would be ideal if I didn't have to use nixUnstable.

**Comment by shlevy** [comment]
Yep, I'll test to see if the latest works against nix stable.

**Comment by shlevy** [comment]
Fixed on master and release-18.03

**Comment by 3noch** [comment]
Thanks!
